database/gorm/test_utils.go

```diff
@@ -2,522 +2,593 @@ package gorm
 
 import (
 	"context"
-	"errors"
+	"fmt"
+	"slices"
 
 	"github.com/goravel/framework/contracts/database"
 	"github.com/goravel/framework/contracts/database/orm"
 	"github.com/goravel/framework/contracts/testing"
 	mocksconfig "github.com/goravel/framework/mocks/config"
+	supportdocker "github.com/goravel/framework/support/docker"
 )
 
-var testContext context.Context
+// Define different test model, to improve the local testing speed.
+// The minimum model only initials one Sqlite and two Postgres,
+// and the normal model initials one Mysql, two Postgres, one Sqlite and one Sqlserver.
+const (
+	TestModelMinimum = iota
+	TestModelNormal
 
-type MysqlDocker struct {
-	MockConfig *mocksconfig.Config
-	Port       int
-	user       string
-	password   string
-	database   string
+	// Switch this value to control the test model.
+	TestModel = TestModelNormal
+)
+
+type TestTable int
+
+const (
+	TestTableAddresses TestTable = iota
+	TestTableAuthors
+	TestTableBooks
+	TestTableHouses
+	TestTablePeoples
+	TestTablePhones
+	TestTableProducts
+	TestTableReviews
+	TestTableRoles
+	TestTableRoleUser
+	TestTableUsers
+	TestTableGoravelUser
+)
+
+var testContext = context.Background()
+
+type TestReadWriteConfig struct {
+	ReadPort  int
+	WritePort int
+
+	// Used by Sqlite
+	ReadDatabase string
 }
 
-func NewMysqlDocker(driver testing.DatabaseDriver) *MysqlDocker {
-	config := driver.Config()
+type testMockDriver interface {
+	Common()
+	ReadWrite(config TestReadWriteConfig)
+	WithPrefixAndSingular()
+}
 
-	return &MysqlDocker{MockConfig: &mocksconfig.Config{}, Port: config.Port, user: config.Username, password: config.Password, database: config.Database}
+type TestQueries struct {
+	mysqlDockers     []testing.DatabaseDriver
+	postgresDockers  []testing.DatabaseDriver
+	sqliteDockers    []testing.DatabaseDriver
+	sqlserverDockers []testing.DatabaseDriver
 }
 
-func (r *MysqlDocker) New() (orm.Query, error) {
-	r.mock()
+func NewTestQueries() *TestQueries {
```

Comment:
Is this internal use only?

Reply:
No, need to be used in other framework packages, but a normal user should not use it.